### PR TITLE
Metricset improvements

### PR DESCRIPTION
**Pull request by ruflin**
* Implement types for redis info
* Convert rtt to micro from nano seconds
* Move error to top level to make counting across all metrics possible
* Move docs to independent file

### CODE REVIEW

**Comment by ruflin** [comment]
@tsg @andrewkroh @brandonmensing I converted nano seconds to micro seconds in rtt.

File: metricbeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -49,7 +49,7 @@ type: long
 
 required: True
 
-Event round trip time time in nano seconds
+Event round trip time in micro seconds
```
Comment (andrewkroh) [inline metricbeat/docs/fields.asciidoc]: `s/micro seconds/microseconds/g`

File: metricbeat/helper/metricset.go [hunk 2]
```diff
@@ -95,12 +89,12 @@ func (m *MetricSet) createEvent(event common.MapStr) common.MapStr {
 
 	event = applySelector(event, m.Config.Selectors)
 
-	// TODO: Add fields_under_root option for "metrics"?
 	event = common.MapStr{
 		"type":                  typeName,
 		eventFieldName:          event,
 		"metricset":             m.Name,
 		"module":                m.Module.name,
+		"rtt":                   rtt.Nanoseconds() / 1000, // Micro seconds
```
Comment (andrewkroh) [inline metricbeat/helper/metricset.go]: `s/1000/time.Microsecond/`, I would prefer the constant to the magic number.

File: metricbeat/docs/fields.asciidoc (hunk 1 above)
Comment (ruflin) [inline metricbeat/docs/fields.asciidoc]: Fixed.

File: metricbeat/helper/metricset.go (hunk 2 above)
Comment (ruflin) [inline metricbeat/helper/metricset.go]: Done

**Pull request merged**